Add blog post on Smart editor and reusing existing config in Platform UI (#1172)
debezium/debezium.github.io: _posts/2025-09-12-debezium-platform-smart-editor.adoc

---
layout: post

> title:  "Debezium Platform: Reusing configuration of a Debezium connector on Kafka Connect or Debezium Server config"

Reviewer: Though the title express exactly the content maybe we can find a more catchy one. @Naros Any suggestion from a native?

Author: Some suggestions from GPT 
"_**Debezium Platform Made Easy: Import and Reuse Existing Connector Configurations**_"
"_**Debezium Platform: Accelerating CDC Pipelines by Reusing Existing Debezium Configurations**_"
"**_Debezium Platform: Simplifying Pipeline Creation with Existing Configurations_**"

Reviewer: What do you think of "Debezium Platform: Simplifying pipeline creation with existing Debezium configurations"?

Author: I like it 👍 clear and communicating the message effectively

Author: updated

Reviewer 2: Sorry for being late to the party, but what about this @mfvitale @indraraj ?
**_Debezium Platform - Create Instant Pipelines with Existing Configurations_**

Author: I like both with a slight preference to the current one i.e "Simplifying Pipeline Creation" WDYT @mfvitale

Reviewer: I like both. so up to you @indraraj

date:   2025-09-02
tags: [ UI, debezium-platform, integration, debezium operator, debezium, debezium-server ]
author: indrashukla
---

Since introducing the Debezium Management Platform (Debezium Platform), our goal has been to make it easy to build CDC data pipelines so you can focus on how your data flows from sources to destinations. Many users already run Debezium connector on Kafka Connect or with Debezium Server. 

> So to further easy the onboarding and jump start the process we introduced the ability to reusing configuration of a any existing Debezium connector on Kafka Connect or any Debezium Server configuration in the Debezium Platform.

Reviewer: ```suggestion
So to further ease the onboarding and jump start the process, we introduced the ability to reuse the configuration of any existing Debezium connector on Kafka Connect or any Debezium Server configuration in the Debezium Platform.
```

Stage UI intelligently recognizes the format of the configuration that you provide, and prompts you to reformat it for use by the platform.

For example, consider the following JSON for specifying the configuration of a Debezium MySQL connector on Kafka Connect:

[source,json,options="nowrap"]
----
{
  "name": "inventory-connector",
  "config": {
    "connector.class": "io.debezium.connector.mysql.MySqlConnector",
    "tasks.max": "1",
    "database.hostname": "mysql",
    "database.port": "3306",
    "database.user": "debezium",
    "database.password": "dbz",
    "database.server.id": "184054",
    "topic.prefix": "dbserver1",
    "database.include.list": "inventory"
  }
}
----

When you add this JSON to the *Smart Editor*, it detects that it is formatted for Kafka Connect and displays an alert.

[.centered-image.responsive-image]
====
++++
<img src="/assets/images/2025-09-12-debezium-platform-smart-editor/Debezium-platform-smart-editor-kafka.png" class="responsive-image">
++++
====

The Smart Editor toolbar then displays an **Auto Format** option, as shown in the preceding figure. 

> The Autoformat option automatically converts the provided JSON into the required Debezium Platform JSON format.

Reviewer: ```suggestion
The `Autoformat` option automatically converts the provided JSON into the required Debezium Platform JSON format.
```

> Optionally, you can edit the JSON to update the value of the `name` field or populate the` description` field.

Reviewer: ```suggestion
Optionally, you can then edit the JSON, for example to update the value of the `name` field or populate the` description` field.
```

> The following image shows the JSON that results after you use the Autoformat option to convert the configuration for a MySQL data source. 

Reviewer: ```suggestion
The following image shows the JSON that results after you use the `Autoformat` option to convert the configuration for a MySQL data source. 
```

[source,properties,options="nowrap"]
----
# ...
debezium.sink.type=pubsub
debezium.sink.pubsub.project.id=debezium-tutorial-local
debezium.sink.pubsub.address=pubsub:8085
debezium.source.connector.class=io.debezium.connector.mysql.MySqlConnector
debezium.source.database.hostname=mysql
debezium.source.database.server.id=223344
debezium.source.database.port=3306
debezium.source.database.user=debezium
debezium.source.database.password=dbz
debezium.source.schema.history.internal=io.debezium.storage.file.history.FileSchemaHistory
debezium.source.schema.history.internal.file.filename=data/schema.dat
debezium.source.offset.storage.file.filename=data/offsets.dat
debezium.source.offset.flush.interval.ms=0
debezium.source.topic.prefix=tutorial
debezium.source.database.include.list=inventory
debezium.source.table.include.list=inventory.customers
# ..
----

When you add the preceding properties file to the Smart Editor, it confirms that it recognizes the file as a Debezium Server configuration.

[.centered-image.responsive-image]
====
++++
<img src="/assets/images/2025-09-12-debezium-platform-smart-editor/debezium-smart-editor-server.png" class="responsive-image">
++++
====

> The Smart Editor toolbar displays an **Auto Format** option, as shown in the preceding image. 

Reviewer: Could you check the other usages of `Auto Format` and use the same for all reference?

Author: sure

Author: updated
